### PR TITLE
Lexical binding

### PR DESCRIPTION
**Pull request by luisgerhorst**

### CODE REVIEW

**Comment by luisgerhorst** [comment]
Since #22 `eclim.el` already requires lexical binding for the callbacks, otherwise you get a void variable error `caller-current-buffer` when eclimd starts. I'm working on enabling lexical binding in the other files too but in that one it is required I think.

File: eclim-common.el [hunk 1]
```diff
@@ -526,9 +526,11 @@ refresh of the problems buffer."
     (if (not (minibuffer-window-active-p (minibuffer-window)))
         (if (string= "e" eclim--problems-filter)
             (message "Eclim reports %d errors." (length problems))
-          (message "Eclim reports %d errors, %d warnings."
-                   (length (cl-remove-if (lambda (p) (null '(assoc-default 'warning p))) problems))
-                   (length (cl-remove-if-not (lambda (p) (null '(assoc-default 'warning p))) problems)))))))
+          (let ((warning-count (length (cl-remove-if-not (lambda (p) (assoc 'warning p))
```
Comment (luisgerhorst) [inline eclim-common.el:529]: Why is always `assoc-default` used to access problems properties? Wouldn't `assoc` be better? Or am I overseeing something here?

**Comment by luisgerhorst** [comment]
That's it I think.

**Comment by nloyola** [comment]
Thanks for the great work!

File: eclim-common.el (hunk 1 above)
Comment (nloyola) [reply eclim-common.el:529]: I'm not sure why they (original authors) coded it that way originally. But it's better that you've replaced it with less code that will be easier to maintain.

**Pull request merged**